### ISSUE TITLE
Progress for --dump-json

### ISSUE THREAD

**Issue by kuerbis** [rg3/youtube-dl#5400]
Would it be possible to add a progress bar when fetching the data for a channel- or a playlist-json-dump (--dump-json)?

**Comment by dstftw** [comment]
This would require tweaking each extractor capable to process channel/playlist. Moreover it's not always possible to know the size of a playlist beforehand to calculate upper bound.

**Comment by Kilobyte22** [comment]
Generally i think what would be beneficial would be if there was an option to stream the progress bar as json. This could be done in a way that you can easily do this without a streaming parser:
```json
[{"percent": 0.1, "eta": 600, "speed": 64839}
,{"percent": 1.3, "eta": 595, "speed": 192624}
...
]
```
To parse, split by line and remove the first char or dump it into a streaming parser. This would allow easier interaction from other programs with youtube-dl

**Comment by royale1223** [comment]
@dstftw  Any news / plans regarding this issue?

**Comment by dstftw** [comment]
Neither.

**Comment by ciffelia** [comment]
@dstftw I'm interesting in this feature and would like to contribute. I think it would be nice to add `--dump-progress-json` option to stream progress in [JSON Lines](http://jsonlines.org/) format (aka LDJSON or NDJSON).
e.g.

```sh
$ youtube-dl --dump-progress-json <URL>
{ "status": "downloading", "downloaded_bytes": 1024, "total_bytes": 12345, ... }
{ "status": "downloading", "downloaded_bytes": 8192, "total_bytes": 12345, ... }
{ "status": "finished", "downloaded_bytes": 12345, "total_bytes": 12345, ... }
```

(JSON entries are the same as those passed to [progress_hooks](https://github.com/ciffelia/youtube-dl/blob/2e20cb36364b91c1d928ce896064fdc7c49e82f8/youtube_dl/YoutubeDL.py#L234-L257))

Is it okay?

**Comment by dstftw** [comment]
No, progress hook is for file downloading only. Also such progress reporting for JSON is useless since it does not does not reflect the actual traffic.
